drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,394 @@
+/* ST Microelectronics LIS2DUX12 3-axis accelerometer driver
+ *
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lis2dux12.pdf
+ */
+
+#define DT_DRV_COMPAT st_lis2dux12
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "lis2dux12.h"
+
+LOG_MODULE_REGISTER(LIS2DUX12, CONFIG_SENSOR_LOG_LEVEL);
+
+static int lis2dux12_set_odr(const struct device *dev, uint8_t odr)
+{
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lis2dux12_md_t mode = {.odr = odr};
+
+	return lis2dux12_mode_set(ctx, &mode);
+}
+
+/**
+ * @brief  Accelerometer full-scale selection.[set]
+ *
+ * @param  ctx      read / write interface definitions
+ * @param  val      change the values of fs in reg CTRL5
+ * @retval          interface status (MANDATORY: return 0 -> no Error)
+ *
+ */
+static int32_t lis2dux12_full_scale_set(stmdev_ctx_t *ctx, uint8_t val)
+{
+	lis2dux12_ctrl5_t reg;
+	int32_t ret;
+
+	ret = lis2dux12_read_reg(ctx, LIS2DUX12_CTRL5, (uint8_t *)&reg, 1);
+
+	if (ret == 0) {
+		reg.fs = (uint8_t)val;
+		ret = lis2dux12_write_reg(ctx, LIS2DUX12_CTRL5, (uint8_t *)&reg, 1);
+	}
+
+	return ret;
+}
+
+static int lis2dux12_set_range(const struct device *dev, uint8_t range)
```

Comment:
This function doesn't seem to be configuring the sensor like its reference implementation:

https://github.com/zephyrproject-rtos/hal_st/blob/fb8e79d1a261fd02aadff7c142729f1954163cf3/sensor/stmemsc/lis2dw12_STdC/driver/lis2dw12_reg.c#L463

might need to take a closer look

Reply:
You're right, I modified the function to mirror the reference. Please see the comment above. https://github.com/zephyrproject-rtos/zephyr/pull/65259#discussion_r1396049864

Reply:
done